### PR TITLE
ref(workflow): Remove loading state from groupEventDetails

### PR DESCRIPTION
**Pull request by scttcper**
We no longer need to block render in this component, all endpoints being loaded are additional details that have their own loading mock.
We also don't need to call fetchData on every event ID change.

This will start showing the event details faster by not showing a spinner while waiting for the `/releases/completion/` endpoint.

@leedongwei I moved `metric.measure` to `componentDidMount`

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx [hunk 1]
```diff
@@ -127,27 +129,11 @@ class GroupDetails extends React.Component<Props, State> {
...
-      switch (err?.status) {
-        case 404:
-          errorType = ERROR_TYPES.GROUP_NOT_FOUND;
-          break;
-        case 403:
-          errorType = ERROR_TYPES.MISSING_MEMBERSHIP;
-          break;
-        default:
-      }
-      this.setState({error: true, errorType, loading: false});
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx:150]: Do we not want to keep this?

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx [hunk 2]
```diff
@@ -132,39 +126,17 @@ class GroupEventDetails extends React.Component<Props, State> {
...
-        error: false,
-        loading: false,
-      });
-    } catch (err) {
-      // This is an expected error, capture to Sentry so that it is not considered as an unhandled error
-      Sentry.captureException(err);
-      this.setState({
-        error: true,
-        loading: false,
-      });
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx:166]: :thunking-face: what was this trying to catch?

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx [hunk 3]
```diff
@@ -132,39 +126,17 @@ class GroupEventDetails extends React.Component<Props, State> {
...
-    }
+    this.setState({releasesCompletion});
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx:139]: Do we need to handle this "loading" state in `<EventEntries>` now that it renders before this is completed?

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx (hunk 1 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx:150]: errorType is used when the group endpoint fails, i had added it here, but it isn't correct. The event error is a different component that doesn't currently care about 404s.

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx (hunk 2 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx:166]: Probably the stuff i deleted in the perf PR...

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx (hunk 3 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx:139]: It has an empty state?

Comment (scttcper) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetails/groupEventDetails.tsx:139]: Could be some pop in

**Review by billyvg — COMMENTED** [review]
Looks like the sidebar gets re-rendered when you go from latest to an event id (e.g. by clicking Older) -- otherwise looks good

**Review by davidenwang — APPROVED** [review]
clicked around and it looks good, feeling 🐇 ✈️ 🐎 🏁 the speed

**Pull request merged**